db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -1539,6 +1539,10 @@ public ResultSetNode optimize(DataDictionary dataDictionary,
         optimizer=getOptimizer(fromList, wherePredicates, dataDictionary, orderByList);
         optimizer.setOuterRows(outerRows);
 
+        // Aggregation with no GROUP BY always outputs one row.
+        if (selectAggregates != null && selectAggregates.size() > 0 && hasNoGroupBy())
```

Comment:
It would be better to check havingAggregates also to cover the case where aggregates is only in having clause but not in the select clause:
if ((selectAggregates != null && !selectAggregates.isEmpty() > 0 || havingAggregates !=null && !havingAggregates.isEmpty()) && hasNoGroupBy())